### PR TITLE
Get locks in tctl get all

### PR DESCRIPTION
**Pull request by flyinghermit**
- Get locked resources in `tctl get all`
- Bootstrap locked resource when starting teleport with `--bootstrap="backup.yml"` flag.

Fixes https://github.com/gravitational/teleport/issues/23765

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

File: lib/services/local/resource.go [hunk 1]
```diff
@@ -350,6 +352,25 @@ func itemsFromLocalAuthSecrets(user string, auth types.LocalAuthSecrets) ([]back
...
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	item := &backend.Item{
+		Key:     backend.Key(locksPrefix, l.GetName()),
+		Value:   value,
+		Expires: l.Expiry(),
+		ID:      l.GetResourceID(),
+	}
+	return item, nil
```
Comment (codingllama) [inline lib/services/local/resource.go:371]: nit:

```suggestion
	return &backend.Item{
		Key:     backend.Key(locksPrefix, l.GetName()),
		Value:   value,
		Expires: l.Expiry(),
		ID:      l.GetResourceID(),
	}, nil
```

File: lib/services/local/resource_test.go [hunk 2]
```diff
@@ -230,3 +230,22 @@ func newUserTestCase(t *testing.T, name string, roles []string, withSecrets bool
...
+			User: "user",
+		},
+		Message: "lock test",
+	})
+	require.NoError(t, err)
+	require.NoError(t, CreateResources(ctx, tt.bk, nl))
+
+	l, err := tt.suite.Access.GetLock(ctx, "test")
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff(nl, l))
```
Comment (codingllama) [inline lib/services/local/resource_test.go:250]: Maybe:

```suggestion
	require.Empty(t, cmp.Diff(nl, l, protocmp.Transform()))
```

(Transform is https://pkg.go.dev/google.golang.org/protobuf/testing/protocmp#Transform)

This avoids occasional failures due to differences in internal (XXX_) fields. protocmp is not meant for Gogo, but still I think it works. Give it a go.

**Review by codingllama — APPROVED** [review]
I have not dug into this before, but seems reasonable to me.

File: lib/services/local/resource_test.go (hunk 2 above)
Comment (sshahcodes) [reply lib/services/local/resource_test.go]: Thanks. added in https://github.com/gravitational/teleport/pull/27168/commits/20e273cc9af83ff98d52ee56381e21389e4b8a1f

**Review by codingllama — APPROVED** [review]

**Pull request merged**